Nos/Views/ReplyCountLabel.swift

```diff
@@ -0,0 +1,154 @@
+import Dependencies
+import SwiftUI
+
+struct ReplyCountLabel: View {
+    
+    var replyCount: ReplyCount
+    var note: Event
+
+    @State private var relaySubscriptions = SubscriptionCancellables()
+    @FetchRequest private var replies: FetchedResults<Event>
+    @Dependency(\.currentUser) var currentUser
+    @EnvironmentObject private var relayService: RelayService
+    @State private var avatars = [URL?]()
+
+    init(replyCount: ReplyCount, for note: Event) {
+        self.note = note
+        self.replyCount = replyCount
+
+        let noteIdentifier = note.identifier ?? ""
+
+        let format = """
+            SUBQUERY(
+                eventReferences,
+                $e,
+                $e.referencedEvent.identifier = %@ AND
+                    ($e.marker = 'reply' OR $e.marker = 'root')
+            ).@count > 0
+        """
+
+        let predicate = NSCompoundPredicate(
+            andPredicateWithSubpredicates: [
+                NSPredicate(format: "kind = 1"),
+                NSPredicate(
+                    format: format,
+                    noteIdentifier,
+                    noteIdentifier
+                ),
+                NSPredicate(format: "deletedOn.@count = 0"),
+                NSPredicate(format: "author.muted = false")
+            ]
+        )
+
+        let fetchRequest = Event.fetchRequest()
+        fetchRequest.includesPendingChanges = false
+        fetchRequest.includesSubentities = false
+        fetchRequest.sortDescriptors = [
+            NSSortDescriptor(keyPath: \Event.identifier, ascending: true)
+        ]
+
+        fetchRequest.predicate = predicate
+        fetchRequest.relationshipKeyPathsForPrefetching = ["author"]
+        _replies = FetchRequest(fetchRequest: fetchRequest)
+    }
+
+    private var isBeingDiscussed: Bool {
+        !replies.isEmpty
+    }
+
+    @MainActor
```

Comment:
This is because socialGraph depends on the main queue. Maybe there are other solutions that avoid this, but given that we expect a low number of replies, I guess this solution doesn't have an impact on performance.

Reply:
In general in the world of async/await it's fine to call long-running `async` functions on the MainActor. Because every time we `await` something on the MainActor it doesn't block the thread - it yields the thread and the MainActor is free to do other things while the awaited function (in this case running on the SocialGraphCache actor) does work elsewhere.